engine/runtime-parser/src/main/java/org/enso/compiler/core/ir/Function.java

```diff
@@ -0,0 +1,144 @@
+package org.enso.compiler.core.ir;
+
+import org.enso.compiler.core.ir.expression.Case;
+import org.enso.persist.Persistance;
+import org.enso.runtime.parser.dsl.GenerateFields;
+import org.enso.runtime.parser.dsl.GenerateIR;
+import org.enso.runtime.parser.dsl.IRChild;
+import org.enso.runtime.parser.dsl.IRField;
+import scala.collection.immutable.List;
+
+public interface Function extends Expression {
+
+  /**
+   * The function arguments.
+   *
+   * <p>Please note that while the source language does not represent multi-argument lambdas, the
+   * internal language can and does.
+   */
+  List<DefinitionArgument> arguments();
+
+  /** The body of the function */
+  Expression body();
+
+  /**
+   * Whether or not the function _can_ be tail-call optimised.
+   *
+   * <p>Please note that this being set to `true` does not _guarantee_ that the function is
+   * optimised.
+   */
+  boolean canBeTCO();
+
+  /** Whether the method is project-private. */
+  boolean isPrivate();
+
+  @GenerateIR(interfaces = {Function.class, IRKind.Primitive.class})
+  final class Lambda extends FunctionLambdaGen {
+    @GenerateFields
+    public Lambda(
+        @IRChild List<DefinitionArgument> arguments,
+        @IRField boolean canBeTCO,
+        @IRChild Persistance.Reference<Expression> body,
```

Comment:
This would become
```suggestion
        @IRChild(deferred=true) Expression body,
```

Reply:
Cannot do that. There must be a field of type `Persistance.Reference` in the generated superclass. See comment in https://github.com/enso-org/enso/pull/12474#discussion_r2008151325